.github/workflows/failureNotifier.yml

```diff
@@ -0,0 +1,96 @@
+name: Notify on Workflow Failure
+
+on:
+  workflow_run:
+    workflows: ["Process new code merged to main"]
+    types:
+      - completed
+
+permissions:
+  issues: write
+
+jobs:
+  notify-failure:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Fetch Workflow Run Jobs
+        id: fetch-workflow-jobs
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const runId = "${{ github.event.workflow_run.id }}";
+            const jobsData = await github.rest.actions.listJobsForWorkflowRun({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: runId,
+            });
+            return jobsData.data;
+      
+      - name: Get merged pull request number
+        id: getMergedPullRequestNumber
+        uses: actions-ecosystem/action-get-merged-pull-request@59afe90821bb0b555082ce8ff1e36b03f91553d9
+        with:
+          github_token: ${{ github.token }}
+
+
+      - name: Process Each Failed Job
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const prNumber = ${{ steps.getMergedPullRequestNumber.outputs.number }};
+            const jobs = ${{ steps.fetch-workflow-jobs.outputs.result }};
+            const failureLabel = 'workflow-failure';
+
+            const prData = await github.rest.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: prNumber,
+            });
+
+            const pr = prData.data;
+
+            const prLink = pr.html_url;
+            const prAuthor = pr.user.login;
+            const prMerger = pr.merged_by.login;
+
+            for (let i = 0; i < jobs.total_count; i++) {
+              if (jobs.jobs[i].conclusion == 'failure') {
+                const jobName = jobs.jobs[i].name;
+                const jobLink = jobs.jobs[i].html_url;
+                const issues = await github.rest.issues.listForRepo({
```

Comment:
Can we search for issues by `jobName` instead? Concerned that looping over 1k+ issues is a bit of a performance hog!